src/core/evaluator.js

```diff
@@ -2178,17 +2229,20 @@ class PartialEvaluator {
     }
 
     function runBidiTransform(textChunk) {
-      var str = textChunk.str.join("");
-      var bidiResult = bidi(str, -1, textChunk.vertical);
+      const text = textChunk.str.join("");
+      const bidiResult = bidi(text, -1, textChunk.vertical);
+      const str = normalizeWhitespace
+        ? replaceWhitespace(bidiResult.str)
+        : bidiResult.str;
       return {
-        str: normalizeWhitespace
-          ? replaceWhitespace(bidiResult.str)
-          : bidiResult.str,
+        str,
         dir: bidiResult.dir,
-        width: textChunk.width,
-        height: textChunk.height,
+        width: textChunk.totalWidth,
+        height: textChunk.totalHeight,
         transform: textChunk.transform,
         fontName: textChunk.fontName,
+        isSpace: AllWhitespaceRegexp.test(str),
```

Comment:
Unless I'm missing something obvious here, in the latest version of the patch it appears that this property is *only* being using (on the main-thread) for the `task._enhanceTextSelection` mode?

If so, it feels a bit unnecessary to run a regular expression on *all* text (for a mode not used by default), and my question is if we cannot instead run the regex-check in the `src/display/text_layer.js` file only when `task._enhanceTextSelection` is actually set?